.github/workflows/release.yml

```diff
@@ -17,13 +17,35 @@ jobs:
       - id: release
         uses: google-github-actions/release-please-action@v4
 
+
       # If release-please created a release, publish to npm
-      - if: ${{ steps.release.outputs.release_created }}
+      - if: steps.release.outputs.release_created
         uses: actions/checkout@v4
-      - if: ${{ steps.release.outputs.release_created }}
+      - if: steps.release.outputs.release_created
         name: actions/setup
         uses: ./.github/actions/setup
+      - if: steps.release.outputs.release_created
+        run: npm pack
+        shell: bash
+      - if: steps.release.outputs.release_created
+        name: Get release version and release package file name
+        run: |
+          PACKAGE_VERSION=$(jq '.version' package.json | tr -d '"')
+          echo "PACKAGE_VERSION=${PACKAGE_VERSION}" >> "$GITHUB_ENV"
+          echo "PACKAGE_FILE=bson-${PACKAGE_VERSION}.tgz" >> "$GITHUB_ENV"
```

Comment:
There's a mixture of variable usage here that I think we can consolidate a bit. So reading the [docs `GITHUB_ENV`](https://docs.github.com/en/actions/learn-github-actions/variables#passing-values-between-steps-and-jobs-in-a-workflow) will set an environment variable for all subsequent steps which means you can access it with `$PACKAGE_FILE` rather than `${{ env.PACKAGE_FILE }}` even if both work one is accessing the env variable "in shell" the other is inserting the value into the "script" string before the shell has been given the "script". 

There's also `GITHUB_OUTPUT` which is what we used in [release automation](https://github.com/mongodb/node-mongodb-native/blob/main/.github/scripts/util.mjs#L5-L20) (The JS there is equivalent to `echo "x=y" >> $GITHUB_OUTPUT`) which allows you to hand the variable to github's runner logic so you can pass values between jobs. Then only the `${{ }}` syntax is the way you can access the variable. 

I prefer the latter, because setting environment variables can have side effects, the ones we are using here probably won't but keeping the values explicitly external seems like a defensive programming approach. But either way, let's go all in on one, so if we're setting env variables then the scripts should use the env variable rather than being generated strings.

Reply:
If we take this further, we can make this all a composite action that takes the variables here as inputs. Even if we don't make a composite action in this PR, writing the yaml to use outputs today is a small lift that will remove the context needed on another day to sort out which variables source from where.

Reply:
I've gone with the latter option; I agree, in the context of future composite actions, it'll make future work easier.